php/libraries/NDB_BVL_InstrumentStatus_ControlPanel.class.inc

```diff
@@ -105,9 +105,14 @@ class NDB_BVL_InstrumentStatus_ControlPanel extends NDB_BVL_InstrumentStatus
 
         if (empty($this->subtest)) {
             $hasAccess = $this->_hasAccess();
-
-            if ($this->getDataEntryStatus() == null && $hasAccess) {
-                $this->setDataEntryStatus('In Progress');
+            if ($this->_status === null && !empty($this->_commentID)) {
+                $this->select($this->_commentID);
+            }
```

Comment:
1. This should be in getDataEntryStatus so that anything that calls the function gets a populated _status property.
2. What happened to the `getDataEntryStatus() === null` part of the if statement?

Reply:
@driusan I moved the code with latest commit. My assumption is the `getDataEntryStatus() === null` isn't necessary since I'm doing:
```
$this->setDataEntryStatus(
                    $this->_status['Data_entry']
                    ?? 'In Progress'
                );
```
and that sets the status to "In Progress" if this->_status['Data_entry'] is null (for keeping the original functionality if first time accessing instrument).